core/CliMulti.php

```diff
@@ -103,6 +108,26 @@ public function __construct(LoggerInterface $logger = null)
         $this->logger = $logger ?: new NullLogger();
     }
 
+    public function handleSignal(int $signal): void
+    {
+        $this->signal = $signal;
+
+        if (\SIGTERM !== $signal) {
+            return;
+        }
+
+        foreach ($this->processes as $process) {
+            if ($process instanceof ProcessSymfony) {
```

Comment:
Merge base has been set to #22513, for easier review and to document requirements for the full implementation.